internal_types/src/entry.rs

```diff
@@ -1087,6 +1119,8 @@ enum ColumnRaw<'a> {
     Bool(Vec<bool>),
 }
 
+pub type ClockValue = u64;
```

Comment:
Love a good descriptive type! You might want to consider making this:

```suggestion
pub struct ClockValue(u64);
```

Or maybe if you need the u64 to be accessible outside this package:
```rust
pub struct ClockValue(pub u64);

// or better yet
impl ClockValue {
    pub fn get(&self) -> u64 { ... }
}
```

Because the type alias `type Thing = u64` doesn't enforce anything - it just becomes "another way of writing u64" ([example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=46d218d88ab109dd68740dbe2a11e130)) and the compiler won't type check your new type like you might expect. With the wrapper type, the compiler will start [checking the types](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=43dcf23a9c9c3268c460f494d2b2fff9) for you.

Reply:
💯

Reply:
Neat